src/Maybe/Extra.elm

```diff
@@ -145,6 +145,20 @@ filter f m =
             Nothing
 
 
+{-| Put a value in a Maybe depending on a predicate
```

Comment:
If we do go forward with this funvtion, I think there might be a way to word this so it's more clear how it'd be used. What do you think about something like
"Keep a value only if the predicate returns `True`. If the predicate fails, returns `Nothing`."

Reply:
That seems reasonable. Before I commit anything, what do you think of:
```
Conditional (guarded) construction of a Maybe.
Keep the value only if the predicate returns True.
Or empty if the predicate returns False.
```